external-crates/move/crates/move-cli/src/base/coverage.rs

```diff
@@ -131,10 +122,119 @@ impl Coverage {
                 disassembler.add_coverage_map(coverage_map.to_unified_exec_map());
                 println!("{}", disassembler.disassemble()?);
             }
-            CoverageSummaryOptions::Lcov => {
+            CoverageSummaryOptions::Lcov { .. } => {
                 unreachable!()
             }
         }
         Ok(())
     }
+
+    pub fn output_lcov_coverage(
+        path: PathBuf,
+        mut config: BuildConfig,
+        differential: Option<String>,
+        test: Option<String>,
+    ) -> anyhow::Result<()> {
+        // Make sure we always compile the package in test mode so we get correct source maps.
+        config.test_mode = true;
+        let package = config.compile_package(&path, &mut Vec::new())?;
+        let units: Vec<_> = package
+            .all_modules()
+            .cloned()
+            .map(|unit| (unit.unit, unit.source_path))
+            .collect();
+        let traces = path.join("traces");
+        let sanitize_name = |s: &str| s.replace("::", "__");
+        let trace_of_test = |test_name: &str| {
+            let trace_substr_name = format!("{}.", sanitize_name(test_name));
+            std::fs::read_dir(&traces)?
+                .filter_map(|entry| {
+                    let entry = entry.unwrap();
+                    let path = entry.path();
+                    if path.is_file()
+                        && path
+                            .file_name()
+                            .unwrap()
+                            .to_str()
+                            .unwrap()
+                            .contains(&trace_substr_name)
+                    {
+                        Some(path)
+                    } else {
+                        None
+                    }
+                })
+            .next()
+                .ok_or_else(|| {
```

Comment:
this is some funky formatting

Reply:
Indeed! And Rust format seems to be happy with it which I find doubly strange since this isn't in a macro.